fix: explain E0120 better cover cases when its raised (#127949)
rust-lang/rust: compiler/rustc_error_codes/src/error_codes/E0120.md

Drop was implemented on a trait object or reference, which is not allowed; only

> structs, enums, and unions can implement Drop.

Reviewer: Ah one other nit, Drop should probably be backticked `` `Drop` ``

}
```

```compile_fail,E0120
struct Concrete {}

impl Drop for &'_ mut Concrete  {
    fn drop(&mut self) {}
}
```

> A workaround for traits is to wrap the trait up in a struct, and implement
> Drop on that:

Reviewer: I think this could be a bit more exact:

```suggestion
A workaround for traits is to create a wrapper struct with a generic type,
add a trait bound to the type, and implement `Drop` on the wrapper:
```

```

> Alternatively, `Drop` can be implemented on trait objects:

Reviewer: Sorry, forgot to clarify it's not implementing the trait object directly. Should be something like this, feel free to reword

```suggestion
Alternatively, the `Drop` wrapper can contain a trait object:
```
